packages/compile-solidity/src/run.ts

```diff
@@ -1,14 +1,81 @@
-const debug = require("debug")("compile:run");
-const OS = require("os");
-const semver = require("semver");
-const Common = require("@truffle/compile-common");
-const { CompilerSupplier } = require("./compilerSupplier");
-const { zeroLinkReferences, formatLinkReferences } = require("./shims");
+import { zeroLinkReferences, formatLinkReferences } from "./shims";
+import debugModule from "debug";
+const debug = debugModule("compile:run");
+import OS = require("os");
+import semver from "semver";
+import { CompilerSupplier } from "./compilerSupplier";
+import type {
+  Compilation,
+  Source,
+  CompiledContract
+} from "@truffle/compile-common";
+import type Config from "@truffle/config";
+import * as Common from "@truffle/compile-common";
+
+type Targets = string[];
+
+type CompilerOutput = {
+  contracts: {
+    [path: string]: object;
+  };
+  sources: {
+    [path: string]: {
+      ast?: object;
+      legacyAST?: object;
+      id: number;
+    };
+  };
+  errors?: any[];
+};
+
+type ProcessAllSourcesArgs = {
+  sources: Common.Sources.Sources;
+  compilerOutput: CompilerOutput;
+  originalSourcePaths: any;
+  language: string;
+};
+
+type PrepareCompilerInputArgs = {
+  sources: Common.Sources.Sources;
+  targets: Targets;
+  language: string;
+  settings: any;
+  modelCheckerSettings: any;
+};
+
+type InternalOptions = {
+  language?: string;
+  noTransform?: boolean;
+  solc?: any;
```

Comment:
Can this be given a more specific type?

Reply:
I'm not certain about this as this is something returned from `solcWrap`. If you follow it back, it returns whatever gets returned [here](https://github.com/trufflesuite/truffle/blob/develop/packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts#L119-L129). I don't know what that is exactly, any suggestions?

Reply:
I'm not too fussed.

Worst case you could define a custom interface here that at least captures the members of the type on which you depend, but I don't see a ton of value in that, as it would make more sense for there to be actual types coming from `solcWrap`.

I don't think there is a good answer to this, but the only lingering concern that I have is over what would happen if there was suddenly proper typing for `solcWrap`? Ideally this dependency would be immediately discoverable due to a compiler error, but that won't happen if it's `any`.

I suppose a custom local type would break if it had structural conflicts with the hypothetical typed return value of `solcWrap`, but I dunno - I don't like this package assuming authority over another package's types.

So yeah, this is probably just a really long-winded way of saying "yeah, fair enough." 😆